cmd/neofs-node/morph.go

```diff
@@ -38,11 +57,33 @@ func initMorphComponents(c *cfg) {
 }
 
 func listenMorphNotifications(c *cfg) {
-	subs, err := subscriber.New(c.ctx, &subscriber.Params{
-		Log:         c.log,
-		Endpoint:    c.viper.GetString(cfgMorphNotifyRPCAddress),
-		DialTimeout: c.viper.GetDuration(cfgMorphNotifyDialTimeout),
+	var (
+		err  error
+		subs subscriber.Subscriber
+	)
+
+	endpoints := c.viper.GetStringSlice(cfgMorphNotifyRPCAddress)
+	timeout := c.viper.GetDuration(cfgMorphNotifyDialTimeout)
+
+	crand := rand.New() // math/rand with cryptographic source
+	crand.Shuffle(len(endpoints), func(i, j int) {
+		endpoints[i], endpoints[j] = endpoints[j], endpoints[i]
 	})
+
+	for i := range endpoints {
```

Comment:
I think we need to add `len(endpoints) == 0` check. Same in `initMorphComponents` func.

Reply:
Fixed.